C++/test/algorithm/backtracking/n_queens.cpp

```diff
@@ -0,0 +1,49 @@
+#include "third_party/catch.hpp"
+#include "algorithm/backtracking/n_queens.hpp"
+
```

Comment:
In general, I think it would be aid readability if you added a comment for each N, as in:

```
// N = 0
NQueensSolver b0(0);
REQUIRE_FALSE(b0.can_place_queens());

// N = 2
NQueensSolver b2(2);
REQUIRE_FALSE(b2.can_place_queens());
```
etc.